src/query/useGitHubReadmeText.ts

```diff
@@ -0,0 +1,40 @@
+import { useQuery } from "@tanstack/react-query";
+
+/**
+ * Fetch readme file from a GitHub repo.
+ */
+export const useGitHubReadmeText = ({
+  repo,
+  commit,
+  isActive,
+}: {
+  repo: string;
+  commit: string;
+  isActive: boolean;
+}) => {
+  const query = useQuery<string | null>({
+    queryKey: ["useGitHubReadmeText", repo, commit],
+    queryFn: async () => {
+      if (!repo || !commit) {
+        return null;
+      }
+
+      try {
+        const response = await fetch(
+          `https://raw.githubusercontent.com/${repo}/${commit}/README.md`,
+        );
+
+        if (response.status !== 200) {
+          return null;
```

Comment:
should we return a message here?

Reply:
We handle this on the UI. If there is no valid response, we show a message "Couldn't find README.md in this repository".